include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -377,8 +379,27 @@ class operational_domain_impl
             output_bdl_pairs{detect_bdl_pairs<Lyt>(
                 layout, sidb_technology::cell_type::OUTPUT,
                 ps.operational_params.input_bdl_iterator_params.bdl_wire_params.bdl_pairs_params)},
-            num_dimensions{params.sweep_dimensions.size()}
+            num_dimensions{params.sweep_dimensions.size()},
+            input_bdl_wires{detect_bdl_wires(lyt, params.operational_params.input_bdl_iterator_params.bdl_wire_params,
+                                             bdl_wire_selection::INPUT)},
+            output_bdl_wires{detect_bdl_wires(lyt, params.operational_params.input_bdl_iterator_params.bdl_wire_params,
+                                              bdl_wire_selection::OUTPUT)}
     {
+        if (lyt.has_cell_type(technology<Lyt>::cell_type::LOGIC))
+        {
+            lyt.foreach_cell(
+                [&](const auto& c)
+                {
+                    if (lyt.get_cell_type(c) == technology<Lyt>::cell_type::LOGIC)
+                    {
+                        canvas_lyt.assign_cell_type(c, technology<Lyt>::cell_type::NORMAL);
+                    }
+                });
+        }
+        else
+        {
+            assert(true && "canvas_lyt is empty");
+        }
```

Comment:
This means you're disallowing the utilization of any layouts without `LOGIC` type cells in the operational domain computation, right? I would strongly advise against this. SiQAD doesn't support the `LOGIC` cell type, thus, we exclude all layouts designed with SiQAD from operational domain computation. Let's not do this!